gapic/schema/api.py

```diff
@@ -560,6 +561,116 @@ def mixin_http_options(self):
             res[s] = [rule for rule in opt_gen if rule]
         return res
 
+    @cached_property
+    def all_methods(self) -> Mapping[str, MethodDescriptorProto]:
+        """Return a map of all methods for the API.
+
+        Return:
+            Mapping[str, MethodDescriptorProto]: A mapping of MethodDescriptorProto
+                values for the API.
+        """
+        return {
+            f"{service_key}.{method_key}": method_value
+            for service_key, service_value in self.services.items()
+            for method_key, method_value in service_value.methods.items()
+        }
+
+    def check_method_settings_validity(
```

Comment:
nit nit suggestion: `def enforce _valid_method_settings`

Reply:
Fixed in 5a2d33b8b160a246945e3752e145878581a351b1